connector/solarwindsentityconnector/connector.go

```diff
@@ -90,28 +99,123 @@ func (s *solarwindsentity) ConsumeMetrics(ctx context.Context, metrics pmetric.M
-		entityTypes := getReverseSortKeys(s.entities) // Will be removed when condition logic is added. Prevents random entity type order
-		for _, k := range entityTypes {
-			entity := s.entities[k]
+		for j := range resourceLog.ScopeLogs().Len() {
+			scopeLog := resourceLog.ScopeLogs().At(j)
+
+			for k := range scopeLog.LogRecords().Len() {
+				logRecord := scopeLog.LogRecords().At(k)
+				tc := ottllog.NewTransformContext(logRecord, scopeLog.Scope(), resourceLog.Resource(), scopeLog, resourceLog)
+				err = processCondition(ctx, eventBuilder, entities, relationships, resourceAttrs, s, &parser, tc)
+				if err != nil {
+					s.logger.Error("Failed to process logs condition", zap.Error(err))
+					return err
+				}
+			}
+		}
+	}
+
+	if eventLogs.LogRecordCount() == 0 {
+		return nil
+	}
+
+	return s.logsConsumer.ConsumeLogs(ctx, eventLogs)
+}
+
+// processCondition evaluates the conditions for entities and relationships.
+// If the conditions are met, it appends the corresponding entity or relationship update event to the event builder.
+func processCondition[C any](
+	ctx context.Context,
+	eventBuilder *internal.EventBuilder,
+	entities []config.EntityEvent,
+	relationships []config.RelationshipEvent,
+	resourceAttrs pcommon.Map,
+	s *solarwindsentity,
+	parser *ottl.Parser[C],
+	tc C) error {
+
+	for _, entityEvent := range entities {
+
+		condition := entityEvent.Conditions
+		seq, err := newConditionSeq(
+			*parser,
+			s.telemetrySettings,
+			condition,
+		)
+
+		if err != nil {
+			s.logger.Error("Failed to create condition sequence", zap.Error(err))
+			return err
+		}
+
+		ok, err := seq.Eval(ctx, tc)
+		if err != nil {
+			s.logger.Error("Failed to evaluate condition", zap.Error(err))
+			return err
+		}
+
+		if ok {
+			entity := s.entities[entityEvent.Type]
 			eventBuilder.AppendEntityUpdateEvent(entity, resourceAttrs)
 		}
+	}
+
+	for _, relationship := range relationships {
+
+		conditions := relationship.Conditions
+		seq, err := newConditionSeq(
+			*parser,
+			s.telemetrySettings,
+			conditions,
+		)
+
+		if err != nil {
+			s.logger.Error("Failed to create condition sequence", zap.Error(err))
+			return err
+		}
 
-		// This will be replaced with actual logic when conditions are introduced
-		for _, relationship := range s.relationships {
+		ok, err := seq.Eval(ctx, tc)
+		if err != nil {
+			s.logger.Error("Failed to evaluate condition", zap.Error(err))
+			return err
+		}
+
+		if ok {
 			eventBuilder.AppendRelationshipUpdateEvent(relationship, resourceAttrs)
 		}
 	}
+	return nil
+}
+
+// newConditionSeq creates a new condition sequence from the provided conditions.
+// If no conditions are provided, it defaults to a sequence that always evaluates to true.
+func newConditionSeq[C any](
+	parser ottl.Parser[C],
+	settings component.TelemetrySettings,
+	conditions []string,
+) (*ottl.ConditionSequence[C], error) {
+	if len(conditions) == 0 || (len(conditions) == 1 && conditions[0] == "") {
```

Comment:
```suggestion
	if len(conditions) == 0 {
```
I would omit (if it is not requirement) the next part for "", it complicates the reading and I think it is not that important. If somebody wants to update entity/relationship let him use empty array.